### ISSUE TITLE
TooltipManager - Possible memory leak

### ISSUE THREAD

**Issue by vincenzomazzeo** [mgarin/weblaf#323]
Hi,
I've noticed that when a component has a tooltip added it's always referenced and the memory is never released.

```java
	private static long lastFree = 0l;
	
	public static void main(String[] args) throws Exception {
		WebLookAndFeel.install();
		
		Image image = ImageIO.read(new File(...));
		
		for (int i = 0; i < 1000; i++) {
			ImageIcon icon = new ImageIcon(image);
			WebDecoratedImage wdi = new WebDecoratedImage(icon);
			wdi.setMinimumSize(new Dimension(icon.getIconWidth(), icon.getIconHeight()));
			wdi.setShadeWidth(5);
			wdi.setCursor(Cursor.getPredefinedCursor(Cursor.HAND_CURSOR));
			wdi.setDrawGlassLayer(false);
			TooltipManager.addTooltip(wdi, null, "Test", TooltipWay.up, (int)TimeUnit.SECONDS.toMillis(2));
			printMemory(i + "");
		}
	}
	
	private static void printMemory(String message) {
		Runtime runtime = Runtime.getRuntime();
		DecimalFormat format = new DecimalFormat("###,###,###");
		long total = runtime.totalMemory();
		long max = runtime.totalMemory();
		long free = runtime.freeMemory();
		System.out.printf("[Total: %s - Max: %s - Occupied: %s - Delta: %s] - %s\n", 
		                  format.format(total), 
		                  format.format(max), 
		                  format.format(total - free), 
		                  format.format(lastFree - free), 
		                  message);
		lastFree = free;
	}
```

I don't know if it's a desired behaviour.

Bye

P.S.: btw thank you for your work!

**Comment by mgarin** [comment]
You were trying this code on v1.28 release? Since I have added a lot of memory leak issue fixes since that version, including multiply fixes for tooltip manager. 

You can find v1.29 pre-release artifacts here:
http://weblookandfeel.com/downloads/prerelease/
this version is far more stable and polished and I strongly recommend using it.

**Comment by vincenzomazzeo** [comment]
Yes, I'm using v1.28 (downloaded from maven central).
Asap I'll try it with the 1.29 release.

**Comment by mgarin** [comment]
Is it any better with the newer version?

**Comment by vincenzomazzeo** [comment]
It seems solved with the version 1.29.

Thank you!

**Comment by mgarin** [comment]
I'll be closing this one as soon as v1.29 is released.
Fix is already available in both active branches.
